pkg/restore/restore.go

```diff
@@ -634,20 +647,33 @@ func (ctx *context) restoreResource(resource, namespace, resourcePath string) (a
 		}
 
 		// necessary because we may have remapped the namespace
-		obj.SetNamespace(namespace)
+		// if the namespace is blank, don't create the key
+		if namespace != "" {
+			obj.SetNamespace(namespace)
+		}
 
 		// add an ark-restore label to each resource for easy ID
 		addLabel(obj, api.RestoreLabelKey, ctx.restore.Name)
 
 		ctx.infof("Restoring %s: %v", obj.GroupVersionKind().Kind, obj.GetName())
-		_, err = resourceClient.Create(obj)
-		if apierrors.IsAlreadyExists(err) {
-			addToResult(&warnings, namespace, err)
+		_, restoreErr := resourceClient.Create(obj)
+		if apierrors.IsAlreadyExists(restoreErr) {
+			equal, err := ctx.objectsAreEqual(obj, resourceClient, restoreErr)
+			// Log any errors trying to resolve the conflict; nil means no conflic
+			if err != nil {
+				addToResult(&warnings, namespace, err)
+			}
+			if equal {
+				ctx.infof("Backup for %s: %v matches cluster version, skipping restore.", obj.GroupVersionKind(), obj.GetName())
+			} else {
+				e := fmt.Errorf("%s and backup does not match.", restoreErr)
```

Comment:
What does this error text end up looking like - do you have an example?

Reply:
Here's a snippet from a restore:

```
Warnings:
  Ark:        <none>
  Cluster:  persistentvolumes "pvc-c5c8bb8c-3758-11e8-a959-42010a960060" already exists and backup does not match.
            persistentvolumes "pvc-f07ce647-336d-11e8-9d79-42010a96006f" already exists and backup does not match.
  Namespaces:
    default:        services "kubernetes" already exists and backup does not match.
    kube-system:    endpoints "kube-controller-manager" already exists and backup does not match.
                    endpoints "kube-scheduler" already exists and backup does not match.
                    services "default-http-backend" already exists and backup does not match.
                    services "heapster" already exists and backup does not match.
                    services "kube-dns" already exists and backup does not match.
                    services "kubernetes-dashboard" already exists and backup does not match.
    nginx-example:  services "my-nginx" already exists and backup does not match.
```

Reply:
Thanks